packages/common/src/projects/HubProjects.ts

```diff
@@ -226,11 +254,97 @@ export async function destroyProject(
 // i.e. delegate to searchContent or just do an item search
 // and convert to IHubProject objects.
 //
-// export async function searchProjects(
-//   filter: Filter<"content">,
-//   opts: IHubSearchOptions
-// ): Promise<ISearchResponse<IHubProject>> {
-//   return _searchContent(filter, opts).then((results) => {
-
-//   })
-// }
+export async function searchProjects(
+  filter: Filter<"content">,
+  options: IHubSearchOptions
+): Promise<ISearchResponse<IHubProject>> {
+  // Scope to Hub Projects
+  const scopingFilter: Filter<"content"> = {
+    filterType: "content",
+    typekeywords: {
+      exact: ["HubProject"],
+    },
+  };
+  // merge filters
+  const projectFilter = mergeContentFilter([scopingFilter, filter]);
+  // expand filter so we can serialize to either api
+  const expanded = expandContentFilter(projectFilter);
+  const so = serializeContentFilterForPortal(expanded);
```

Comment:
This entire search section seems to be generalizable for other search methods

Reply:
Agreed - once Tom's content work is finalized, I was planning to re-compose this. What's in here now is basically the internals of `_searchContent` but instead of converting things into `IHubContent` pojos, this converts things into `IHubProject` pojos. I wanted to go through this once without Tom's code, just to see what was really needed, and then look at both and hoist out some central common code that's called from`HubContentManager.search(...)` and the other Entity Manager's.

The other thing `.search` brings up is the idea of forcing `N+1` so that "full" `IHub{Entity}` objects are returned from the function. I believe we'd decided to start like this and then look for optimizations should the performance be unacceptable...